File: api/client/src/main/java/su/plo/lib/api/client/event/render/HudRenderEvent.java
package su.plo.lib.api.client.event.render;

import lombok.Getter;
import org.jetbrains.annotations.NotNull;
import su.plo.lib.api.client.gui.GuiRender;
import su.plo.voice.api.event.Event;

import static com.google.common.base.Preconditions.checkNotNull;

public final class HudRenderEvent implements Event {

    @Getter
    private final GuiRender render;
    @Getter
    private final float delta;

    public HudRenderEvent(@NotNull GuiRender render, float delta) {
        this.render = checkNotNull(render);
        this.delta = delta;
    }
}

File: client/src/main/kotlin/su/plo/voice/client/event/HudRenderEvent.kt
package su.plo.voice.client.event

import su.plo.lib.mod.client.render.gui.GuiRenderContext
import su.plo.slib.api.event.GlobalEvent
import su.plo.voice.client.event.HudRenderEvent.Callback

object HudRenderEvent : GlobalEvent<Callback>(
    { callbacks ->
        Callback { context, delta ->
            callbacks.forEach { it.onRender(context, delta) }
        }
    }
) {

    fun interface Callback {
        fun onRender(context: GuiRenderContext, delta: Float)
    }
}
